Indexing/ResultSubstitution.hpp

```diff
@@ -133,67 +133,12 @@ class ResultSubstitution
    */
   virtual bool isIdentityOnResultWhenQueryBound() {return false;}
 
-  virtual RobSubstitution* tryGetRobSubstitution() { return 0; }
```

Comment:
Could we please keep `tryGetRobSubstitution` as it was? I am using it in https://github.com/vprover/vampire/tree/dump_proof_details when generating `proof_extra` for Anja

Reply:
Actually, if there is a principled solution under way, we don't need to rush it. (The proof_details branch may stay where it is---under current master---until we get around to merging `cleanup-result-substitution`.)

However, let me state what we need there. We don't care about getting a `RobSubstitution` per se. We care about being able to print the bindings (a part of the substitution) relevant, separately, for the query and the result term. Ideally, I would like `ResultSubstitution` to offer such functions. (although I can imagine going through the pain of collecting the variables from, e.g., the query, supplying them as arguments to a dummy term created just for the occasion, calling the respective apply function and reading of the result.) Any thoughts/ideas on this?

Reply:
Well so I think it's not an issue to implement this for other substitutions as well. All of them store all the bound variables in some sort of hashmap, anyways so we can easily iterate over that map and then use the relevant apply function as u say. As you need this only in proof outputing i guess there's not a big issue about performance, right?

But in general there's quite a lot of issues arount the `ResultSubstitution` interface, so i wanna completely delete it and get a new one that is more planned and less grown. But we can definitely introduce a function that allows for something like that easily in a new one.

Reply:
Perfect, let's talk about the details later!